users/tests.py

```diff
@@ -450,3 +454,85 @@ def test_user_profile_delete_unauthorized(self):
         response = self.client.post(url, content_type='application/json')
 
         self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class UserConfirmEmailTest(BaseSingleUserTest):
+    """Tests the confirm_email endpoint. """
+
+    def setUp(self):
+        super().setUp()
+
+        user = User.objects.get(username=self._user['username'])
+        user.person.email_confirmed = False
+        user.person.save(update_fields=['email_confirmed'])
+
+        self._email_confirmation_token = user.person.emailconfirmationtoken
+
+    def test_confirm_email(self):
+        """Test if it possible to confirm email. """
+
+        url = reverse('confirm-email', kwargs={'version': 'v1'})
+        data = {'token': str(self._email_confirmation_token)}
+        response = self.client.post(url, data=json.dumps(data),
+                                    content_type='application/json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_confirm_email_with_invalid_token(self):
+        """Tests if it's not possible to confirm email with invalid token"""
+
+        url = reverse('confirm-email', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps({'token': 'random_token'}),
+                                    content_type='application/json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_confirm_email_with_already_confirmed_email(self):
+        """Tests if it's not possible to confirm email with already confirmed email. """
+
+        user2 = {
+            'username': 'test.user2',
+            'password': 'secret',
+            'email': 'test.user2@domain.com',
+        }
+        user2 = User.objects.create_user(**user2)
+
+        token = str(user2.person.emailconfirmationtoken)
```

Comment:
Don't create an additional user, use user created in `BaseSingleUserTest` class.